src/ContainerInfo.h

```diff
@@ -73,6 +74,33 @@ ContainerTypeEnum containerTypeEnumFromStr(std::string& str);
 const char* containerTypeEnumToStr(ContainerTypeEnum ty);
 
 struct ContainerInfo {
+  struct Codegen {
+    std::string decl;
+    std::string func;
+  };
+
+  ContainerInfo(const ContainerInfo&) = delete;
```

Comment:
Also delete the copy assignment constructor?

https://en.cppreference.com/w/cpp/language/rule_of_three

Reply:
Always useful having someone who knows what they're doing review C++ :)

I'm a little confused now though as I'm almost certain I don't need a custom destructor, I just want to avoid accidental expensive copies. Is this me misreading "requires" from the docs?

Reply:
Oh, I think this bit explains it:

> Classes that manage non-copyable resources through copyable handles may have to declare copy assignment and copy constructor private and not provide their definitions or define them as deleted. This is another application of the rule of three: deleting one and leaving the other to be implicitly-defined will most likely result in errors.